src/tests/ftest/control/daos_scm_config.py

```diff
@@ -0,0 +1,105 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+from apricot import TestWithServers
+from pydaos.raw import DaosApiError
+from command_utils import CommandFailure
+
+
+class SCMConfigTest(TestWithServers):
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+    Simple test to verify the SCM storage config.
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a SCMConfigTest object."""
+        super(SCMConfigTest, self).__init__(*args, **kwargs)
+        self.obj = None
+
+    def write_data(self, data):
+        """Write data obj to a container.
+
+        Args:
+            data (str): string of data to be written.
+        """
+        try:
+            # create an object and write some data into it
+            self.obj = self.container.container.write_an_obj(data,
+                                                             len(data) + 1,
+                                                             "dkey",
+                                                             "akey",
+                                                             obj_cls="OC_S1")
+            self.obj.close()
+            self.log.info("==>    Wrote an object to the container")
+        except DaosApiError as error:
+            self.fail("Test failed during the object write.\n{0}".format(error))
+
+    def test_scm_in_use_basic(self):
+        """
+        JIRA ID: DAOS-2972
+
+        Test Description: Verify that an attempt to configure devices that have
+        already been configured and are in use by DAOS is handled.
+
+        :avocado: tags=all,small,pr,hw,scm_in_use,basic
+        """
+        # Create pool and container
+        self.prepare_pool()
+        self.add_container(self.pool)
+
+        # now open the container and write some data
+        self.container.open()
+        data_w = "Ehrm... Testing... Testing"
+        self.write_data(data_w)
+
+        # Run storage prepare
+        if self.server_managers[-1].manager.job.using_dcpm:
+            # Storage prepare should return error
```

Comment:
I'm confused about this comment. Is Line 84 supposed to return error and raise an exception? If that's the expectation, why do we have self.fail in Line 86?

Reply:
Good catch, I should remove that comment. It's not relevant to the test case, I must've copy pasted it and did not modify it.